compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp

```diff
@@ -61,31 +61,38 @@ void compute_sym_scale_zp(float min, float max, float &scaling_factor, int64_t &
 void compute_asym_scale_zp(float min, float max, float &scaling_factor, int64_t &zp,
                            float &nudged_min, float &nudged_max)
 {
-  assert(min != max);
 
+  LOGGER(l);
   const int32_t kMinScale = 0;
   const int32_t kMaxScale = 255;
   const double qmin_double = kMinScale;
   const double qmax_double = kMaxScale;
   const double rmin = std::fmin(0, min);
   const double rmax = std::fmax(0, max);
-
   double scale = (rmax - rmin) / (qmax_double - qmin_double);
-  const double zero_point_from_min = qmin_double - rmin / scale;
-  const double zero_point_from_max = qmax_double - rmax / scale;
-  const double zero_point_from_min_error = std::abs(qmin_double) + std::abs(rmin / scale);
-  const double zero_point_from_max_error = std::abs(qmax_double) + std::abs(rmax / scale);
-  const double zero_point_double = zero_point_from_min_error < zero_point_from_max_error
-                                       ? zero_point_from_min
-                                       : zero_point_from_max;
-  uint8_t nudged_zero_point = 0;
+  double zero_point_double{0};
+  if (scale == 0)
+  {
+    WARN(l) << "The minimum and maximum values are the same." << std::endl;
+    if (min >= 0 && max >= 0)
+      zero_point_double = kMinScale;
+    else
+      zero_point_double = kMaxScale;
+  }
+  else
+    zero_point_double = qmin_double - rmin / scale;
+  uint8_t nudged_zero_point{0};
   if (zero_point_double <= qmin_double)
   {
```

Comment:
Can you add the assertion for each case?

`assert(min >= 0 && max >= 0);`
`assert(min < 0 && max < 0);`
`assert(min < 0 && max >= 0);`